crates/subspace-networking/src/node_runner.rs

```diff
@@ -112,6 +119,53 @@ impl NodeRunner {
                 },
                 _ = self.networking_parameters_registry.run().fuse() => {
                     trace!("Network parameters registry runner exited.")
+                },
+                _ = &mut self.peer_dialing_timeout => {
```

Comment:
I think this should be combined with `random_query_timeout`, which tries to find peers on the network. Overall the idea I had was to monitor number of connections and respond to disconnections rather than having an interval running every 3 seconds.

Reply:
Added TODO as agreed